cups/generic.py

```diff
@@ -0,0 +1,124 @@
+from cups import _cups
+from cups.enums.http import HttpEncryption
+from cups.types.cups import cupsDest
+from cups.types.ipp import IPPStatus
+from cups.utils import _bytes_to_value, _value_to_bytes
+from typing import Dict, Optional, Tuple
+from pathlib import Path
+
+_ffi = _cups.ffi
+_lib = _cups.lib
+
+
+def areCredentialsValidForName(common_name: str, credentials: str) -> bool:
+    ok = _lib.cupsAreCredentialsValidForName(common_name.encode(), credentials.encode())
+    return bool(_bytes_to_value(ok))
+
+
+def copyCredentialsKey(
+    path: Path,
+    common_name: str,
+    is_key: bool = False,
+    is_public_key: bool = False,
+    copy_req: bool = False,
+) -> str:
```

Comment:
Best practices recommend requiring a name for any bool values:

```suggestion
def copyCredentialsKey(
    path: Path,
    common_name: str,
    *,
    is_key: bool = False,
    is_public_key: bool = False,
    copy_req: bool = False,
) -> str:
```

Ruff and other linters have a rule for this, I don't recall the name of it off the top of my head.

Reply:
Thanks a lot! I didn't know about this logic of making any keywords in kwargs

Reply:
Nice, yeah I like the philosophy of this one. `foo(is_key=True, is_public_key=True)` is easier to read and harder to accidentally set the wrong values compared to `foo(True, False`).